Tasks/Gradle/gradletask.ts

```diff
@@ -124,6 +126,19 @@ function processCodeAnalysisResults(): Q.Promise<void> {
     return sqGradle.uploadSonarQubeBuildSummaryIfEnabled();
 }
 
+// Configure the maximum memory available to Gradle if the user hasn't done so, otherwise it may run ouf of heap space
+function configureMemoryLimits() {
+    if (!process.env['GRADLE_OPTS']) {
```

Comment:
The only possible problem I see with this is if the user can configure the heap size differently (e.g. via the build.gradle file) and we would override it. I couldn't find a way to do this though.